### PR TITLE
Document icu_calendars

### PR DESCRIPTION
**Pull request by Manishearth**
Progress on https://github.com/unicode-org/icu4x/issues/1018

Leftover work from https://github.com/unicode-org/icu4x/pull/1038

### CODE REVIEW

File: components/calendar/src/calendar.rs [hunk 1]
```diff
@@ -12,17 +12,27 @@ use crate::{Date, DateDuration, DurationUnit, Iso};
...
     /// Construct the date from an ISO date
     fn date_from_iso(&self, iso: Date<Iso>) -> Self::DateInner;
+    /// Obtain an ISO date from this date
     fn date_to_iso(&self, date: &Self::DateInner) -> Date<Iso>;
     // fn validate_date(&self, e: Era, y: Year, m: MonthCode, d: Day) -> bool;
     // // similar validators for YearMonth, etc
 
     // fn is_leap<A: AsCalendar<Calendar = Self>>(&self, date: &Date<A>) -> bool;
+    /// Count the number of months in a given year, specified by providing a date
+    /// from that year
```
Comment (sffc) [inline components/calendar/src/calendar.rs:26]: Thought: I'd love if we could give examples for these (although that would require implementing something like the Julian calendar to make it interesting)

File: components/calendar/src/duration.rs [hunk 2]
```diff
@@ -37,6 +48,7 @@ impl<C: Calendar + ?Sized> Default for DateDuration<C> {
 }
 
 impl<C: Calendar + ?Sized> DateDuration<C> {
+    /// Construct a DateDuration
```
Comment (sffc) [inline components/calendar/src/duration.rs:51]: Nit: In some cases, like this one, you can add an example.

File: components/calendar/src/duration.rs [hunk 3]
```diff
@@ -9,18 +9,29 @@ use core::marker::PhantomData;
...
+    /// The number of days
     pub days: i32,
+    /// A marker for the calendar
     pub marker: PhantomData<C>,
 }
 
+/// A "duration unit" used to specify the minimum or maximum duration of time to
+/// care about
 #[derive(Copy, Clone, Eq, PartialEq, Debug)]
 pub enum DurationUnit {
```
Comment (sffc) [inline components/calendar/src/duration.rs:27]: Nit:

```suggestion
pub enum DateDurationUnit {
```

File: components/calendar/src/lib.rs [hunk 4]
```diff
@@ -4,6 +4,15 @@
 
 #![cfg_attr(not(any(test, feature = "std")), no_std)]
 
+/// The `icu_calendars` crate contains the core types used by ICU4X for dealing
```
Comment (sffc) [inline components/calendar/src/lib.rs:7]:
```suggestion
//! The `icu_calendars` crate contains the core types used by ICU4X for dealing
```

**Review by sffc — COMMENTED** [review]
OK, but can you add some examples?

File: components/calendar/src/calendar.rs (hunk 1 above)
Comment (Manishearth) [reply components/calendar/src/calendar.rs:26]: Note that the Calendar trait isn't really the public API, the Date type is. But I'll add examples there later.

**Review by sffc — APPROVED** [review]

**Pull request merged**